app/models/product.rb

```diff
@@ -1,8 +1,30 @@
 class Product < ApplicationRecord
-  validates :title, :description, :image_url, presence: { message: "can't be blank"}
+  class PriceValidator < ActiveModel::Validator
+    def validate(record)
+      unless record.price > record.discount_price
+        record.errors.add :price, "Discount Price Can't be greater than Original Price!"
+      end
+    end
+  end
+  class UrlValidator < ActiveModel::Validator
+    def validate_each(record, attribute, value)
+      unless value =~ %r{\.(gif|jpg|png)\z}i
+        record.errors.add(attribute,
+          (options[:message] || "is not an approriate Image URL format")
+        )
+      end
+    end
+  end
+  # validates_with PriceValidator
+  validates :url, presence: true, url: true
+  validates :title, :description, presence: { message: "can't be blank"}
```

Comment:
what's the default message?

Reply:
The default message is itself  "can't be blank" , so we can remove the message here